### ISSUE TITLE
Integration fails

### ISSUE THREAD

**Issue by Talonpoint** [jdeath/Hubspace-Homeassistant#103]
I have been using the old version and tried to upgrade and everything stopped working. I have tried to troubleshoot on my own, including deleting and reinstalling the custom repository, trying to go back to old version, many Home Assistant restarts. I get to the stage where I try to Add Integration, provide my username and password and submit. At that point Home Assistant generates the following log entry and the integration returns an Unknown error occurred message.

The Home Assistant Logs keep returning the following when I try to add the integration.

Logger: aiohttp.server
Source: /usr/local/lib/python3.12/site-packages/aiohttp/web_protocol.py:433
First occurred: 7:16:52 PM (4 occurrences)
Last logged: 7:40:01 PM

Error handling request
Traceback (most recent call last):
  File "/usr/local/lib/python3.12/site-packages/aiohttp/web_protocol.py", line 462, in _handle_request
    resp = await request_handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/aiohttp/web_app.py", line 537, in _handle
    resp = await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/aiohttp/web_middlewares.py", line 114, in impl
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/security_filter.py", line 92, in security_filter_middleware
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/forwarded.py", line 83, in forwarded_middleware
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/request_context.py", line 26, in request_context_middleware
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/ban.py", line 85, in ban_middleware
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/auth.py", line 242, in auth_middleware
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/headers.py", line 32, in headers_middleware
    response = await handler(request)
               ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/http.py", line 73, in handle
    result = await handler(request, **request.match_info)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/decorators.py", line 81, in with_admin
    return await func(self, request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/config/config_entries.py", line 222, in post
    return await super().post(request, flow_id)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/data_validator.py", line 74, in wrapper
    return await method(view, request, data, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/data_entry_flow.py", line 122, in post
    result = await self._flow_mgr.async_configure(flow_id, data)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/data_entry_flow.py", line 366, in async_configure
    result = await self._async_configure(flow_id, user_input)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/data_entry_flow.py", line 413, in _async_configure
    result = await self._async_handle_step(
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/data_entry_flow.py", line 527, in _async_handle_step
    if result.get("preview") is not None:
       ^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'

**Comment by Expl0dingBanana** [comment]
Thats an interesting one. What version of HA are you running?

**Comment by Talonpoint** [comment]
Home Assistant 
Core 2024.8.3
Frontend 20240809.0
I am running it in a Docker installation on a Synology Diskstation running DSM 7.1.1-42962 Update 6
The Hubspace integration was working fine until the August updates. I just got around to testing and troubleshooting today.

**Comment by Expl0dingBanana** [comment]
What docker image are you using for HA? I have done all my testing from a Synology NAS using `docker.io/linuxserver/homeassistant:latest@sha256:12289b32df4eec21e727c60a87599a047138d502889497e70afa67188fe17505` without any issues.

**Comment by Expl0dingBanana** [comment]
I was able to reproduce the issue with the following steps:

 * Add HubSpace-HomeAssistant to HACs
 * Add the integration HubSpace-HomeAssistant
 * Enter an incorrect login
 * Observe the issue above

**Comment by Expl0dingBanana** [comment]
I believe this has been fixed with #104  . Please review the testing instructions within that PR and post any comments within that PR.

**Comment by Talonpoint** [comment]
This makes sense, since I tried using an incorrect password also and got the same result. However, in my case it didn’t prove/disprove anything since I am using the same UserID and Password that the old system used successfully and I can still use on the iOS Hubspace app. To test the system, I reset my password on Hubspace. I then tried to Add the Integration and use the new userid and password. The integration returned the same error. The new password immediately worked on my iOS app.Regards, RicardoRicardo A. Venegas(732) 829-5891On Sep 1, 2024, at 08:30, Christopher Dohmen ***@***.***> wrote:﻿
I was able to reproduce the issue with the following steps:

Add HubSpace-HomeAssistant to HACs
Add the integration HubSpace-HomeAssistant
Enter an incorrect login
Observe the issue above

**Comment by Expl0dingBanana** [comment]
I am wondering if this is something with the loginor config flow.  Lets manually run the login to see what happens

* Get access to a python3.10 or above environment
  * You can use the HomeAssistant if you can access terminal
  * Install python3.12 locally
   *  pip install "hubspace-async==0.3.0"
 * Run `python3.12`
 * Execute the following within the python terminal and replace username / password
 ```python
 
from hubspace_async import connection
import asyncio

conn = connection.HubSpaceConnection("<username>", "<password>")
loop = asyncio.get_event_loop()
loop.run_until_complete(conn.get_account_id())
```

**Comment by cjr222** [comment]
Ok, this could very well be my lack of python knowledge but when trying to run this code interactively this is what it looks like. Yes, I did replace my username and password, just replaced here for security.
`330f1202f9d0:/config# python
Python 3.12.4 (main, Jun 22 2024, 12:49:21) [GCC 13.2.1 20240309] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from hubspace_async import connection
>>> import asyncio
>>> 
>>> conn = connection.HubSpaceConnection("<username>", "<password>")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python3.12/site-packages/hubspace_async/connection.py", line 55, in __init__
    self.client = websession or ClientSession()
                                ^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/aiohttp/client.py", line 294, in __init__
    loop = loop or asyncio.get_running_loop()
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^
RuntimeError: no running event loop
>>> loop = asyncio.get_event_loop()
<stdin>:1: DeprecationWarning: There is no current event loop
>>> loop.run_until_complete(conn.get_account_id())
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
NameError: name 'conn' is not defined
>>> `

**Comment by Expl0dingBanana** [comment]
No worries! Thw interactive terminal can be very weirs. Can you try running
one command at a time? Sometimes when multiple commands are passed together
it gets funky.

**Comment by Expl0dingBanana** [comment]
I have created an easier way to run the tests by adding a CLI tool to hubspace-async. Here is the updated commands


* Get access to a python3.12 or above environment
   * You can use the HomeAssistant if you can access terminal
   * Run locally
      * Install python3.12 locally
         * https://www.python.org/downloads/release/python-3125/ 
      * Install hubspace-async 
         *  `pip install "hubspace-async==0.4.0[cli]"`
 * Run the following command
 
 ```bash

python -m hubspace_async.cli --username "<username>" --password "<password>" auth-flow
```

If the auth is successful, you should not see any error messages. If that works, lets try to connect to gather the account ID.
To test connectivity, run the following command:
```bash

python -m hubspace_async.cli --username "<username>" --password "<password>" auth-flow
```

If successful, an account id with several hyphens will appear.

**Comment by cjr222** [comment]
Both of those commands end with Token has been successfully generated

**Comment by Expl0dingBanana** [comment]
Well that's a good start! At least we can focus on the configflow

**Comment by allstarnz** [comment]
FWIW I have similar issues as OP. I too was able to generate both tokens.  This integration was working flawlessly until recently.  It stopped working for a while and then started again. Now it doesn't work at all. 

I am still on HACS 1.34.0 I don't know if this matters?

**Comment by allstarnz** [comment]
> FWIW I have similar issues as OP. I too was able to generate both tokens. This integration was working flawlessly until recently. It stopped working for a while and then started again. Now it doesn't work at all.
> 
> I am still on HACS 1.34.0 I don't know if this matters?

And of course, it randomly starts working again 🤔

**Comment by Expl0dingBanana** [comment]
@Talonpoint  Can yall redownload v3.1.1 and try again? This is a weird once since im not able to replicate the issue in the same way yall are experiencing it.

Goto hacs -> Hubspace Homeassistant -> Update Information -> Redownload -> select v3.1.1

**Comment by Expl0dingBanana** [comment]
@allstarnz a previous dev version of 3.1.1 had an issue where HA had to be restarted for it to start working. I wonder if thats what you experienced.

**Comment by engyew** [comment]
> @Talonpoint Can yall redownload v3.1.1 and try again? This is a weird once since im not able to replicate the issue in the same way yall are experiencing it.
> 
> Goto hacs -> Hubspace Homeassistant -> Update Information -> Redownload -> select `fix-config-flow`

I redownloaded 3.1.0 and signed in again, and it now works.  I gonna stay on this version

**Comment by Expl0dingBanana** [comment]
This may be solved with 3.2.0 + latest HA. Please let me know if the problem still persists!

**Comment by Expl0dingBanana** [comment]
Closing due to inactivity. Please re-open the issue if you still encounter the problem.
